app/youtube-grabber.js

```diff
@@ -22,6 +22,29 @@ class YoutubeGrabber {
     const decideResponse = await ytGrabHelp.decideUrlRequestType(channelId, 'channels?flow=grid&view=0&pbj=1', channelIdType)
     const channelPageResponse = decideResponse.response
 
+    const headerLinks = channelPageResponse.data[1].response.header.c4TabbedHeaderRenderer.headerLinks
+    let links = {
+      primaryLinks: [],
+      secondaryLinks: []
+    }
+    if (typeof headerLinks !== 'undefined') {
+      links = headerLinks.channelHeaderLinksRenderer
+      links.primaryLinks = links.primaryLinks.map(x => {
+        return {
+          url: decodeURIComponent(x.navigationEndpoint.urlEndpoint.url.match('&q=(.*)')[1]),
+          icon: x.icon.thumbnails[0].url,
+          title: x.title.simpleText
+        }
+      })
+      links.secondaryLinks = links.secondaryLinks.map(x => {
+        return {
+          url: decodeURIComponent(x.navigationEndpoint.urlEndpoint.url.match('&q=(.*)')[1]),
+          icon: x.icon.thumbnails[0].url,
+          title: x.title.simpleText
+        }
+      })
+    }
```

Comment:
Would be good to separate variables for result and temp storage (for readability or something)
`links` is for result and `channelHeaderLinksData` is temp storage
```suggestion
    const links = {
      primaryLinks: [],
      secondaryLinks: []
    }
    if (typeof headerLinks !== 'undefined') {
      const channelHeaderLinksData = headerLinks.channelHeaderLinksRenderer
      links.primaryLinks = channelHeaderLinksData.primaryLinks.map(x => {
        return {
          url: decodeURIComponent(x.navigationEndpoint.urlEndpoint.url.match('&q=(.*)')[1]),
          icon: x.icon.thumbnails[0].url,
          title: x.title.simpleText
        }
      })
      links.secondaryLinks = channelHeaderLinksData.secondaryLinks.map(x => {
        return {
          url: decodeURIComponent(x.navigationEndpoint.urlEndpoint.url.match('&q=(.*)')[1]),
          icon: x.icon.thumbnails[0].url,
          title: x.title.simpleText
        }
      })
    }
```